pkg/generator/certificate.go

```diff
@@ -339,8 +337,10 @@ func (kp *CertKeyPair) LoadFromData(data map[string][]byte) {
 	switch priv := privateKey.(type) {
 	case *ecdsa.PrivateKey:
 		kp.Cert.PrivateKeyEC = priv
+		kp.V1Spec.Algorithm = v1alpha1.AlgorithmTypeECDSAWithSHA256
```

Comment:
take a look at https://github.com/ForgeRock/secret-agent/pull/134/files#diff-675de4ffdac7698a359e45836ac5051fL428

rootCA is hardcoded to ECDSA. I think we could make this the default in the mutating webbook if the user does not specify anything. However, let the user specify RSA if they want to. 🤷 ???

Reply:
I added the hook